txpool/src/containers/dependency.rs

```diff
@@ -0,0 +1,482 @@
+                    {
+                        Some(contract_id)
+                    } else {
+                        panic!("InputIndex for contract should be always correct");
+                    }
+                }
+                Output::ContractCreated { contract_id } => Some(*contract_id),
+            };
+            let utxo_id = UtxoId::new(tx.id(), index as u8);
+            if let Some(contract_id) = is_contract {
+                // insert contract
+                self.contracts.insert(
+                    contract_id,
+                    ContractState {
+                        depth: max_depth,
+                        used_by: HashSet::new(),
+                        origin: Some(utxo_id),
+                        gas_price: tx.gas_price(),
+                    },
+                );
+            } else {
+                // insert output coin inside by_coin
+                self.coins.insert(
+                    utxo_id,
+                    CoinState {
+                        is_spend_by: None,
+                        depth: max_depth,
+                    },
+                );
+            }
+        }
+
+        Ok(removed_tx)
+    }
+
+    pub fn recursively_remove_all_dependencies<'a>(
+        &'a mut self,
+        txs: &'a HashMap<TxId, ArcTx>,
+        tx: ArcTx,
+    ) -> Vec<ArcTx> {
+        let mut removed_tx = vec![tx.clone()];
+
+        // for all outputs recursivly call removal
+        for (index, output) in tx.outputs().iter().enumerate() {
+            // get contract_id or none if it is coin
+            let is_contract = match output {
+                Output::Coin { .. }
+                | Output::Withdrawal { .. }
+                | Output::Change { .. }
+                | Output::Variable { .. } => None,
+                Output::Contract { input_index, .. } => {
+                    if let Input::Contract { contract_id, .. } = tx.inputs()[*input_index as usize]
+                    {
+                        Some(contract_id)
+                    } else {
+                        panic!("InputIndex for contract should be always correct");
+                    }
+                }
+                Output::ContractCreated { contract_id } => Some(*contract_id),
+            };
+            // remove contract childrens.
+            if let Some(ref contract_id) = is_contract {
+                let state = self.contracts.get(contract_id).cloned();
+                if let Some(state) = state {
+                    // call removal on every tx;
+                    for rem_tx in state.used_by.iter() {
+                        let rem_tx = txs.get(rem_tx).expect("Tx should be present in txs");
+                        removed_tx
+                            .extend(self.recursively_remove_all_dependencies(txs, rem_tx.clone()));
+                    }
+                } else {
+                    panic!("Contract state should be present when removing child");
```

Comment:
ahhh more panics

Reply:
"Contract state should be present when removing child" this is not expected behaviour